pkg/service/providercacher/cachingqueuepoller.go

```diff
@@ -0,0 +1,191 @@
+package providercacher
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+)
+
+const (
+	defaultPollInterval = 5 * time.Second
```

Comment:
Alternatively you could set `WaitTimeSeconds` to the max and not have an interval (or a very short one)...

https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-short-and-long-polling.html#sqs-long-polling